rotkehlchen/db/dbhandler.py

```diff
@@ -751,6 +761,118 @@ def delete_aave_data(self) -> None:
         self.conn.commit()
         self.update_last_write()
 
+    def add_adex_events(
+            self,
+            events: Sequence[Union[Bond, Unbond, UnbondRequest, ChannelWithdraw]],
+    ) -> None:
+        query = (
+            """
+            INSERT INTO adex_events (
+                tx_hash,
+                address,
+                identity_address,
+                timestamp,
+                type,
+                pool_id,
+                amount,
+                usd_value,
+                bond_id,
+                nonce,
+                slashed_at,
+                unlock_at,
+                channel_id
+            )
+            VALUES (?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?)
+            """
+        )
+        cursor = self.conn.cursor()
+        for event in events:
+            event_tuple = event.to_db_tuple()
+            try:
+                cursor.execute(query, event_tuple)
+            except sqlcipher.IntegrityError:  # pylint: disable=no-member
+                self.msg_aggregator.add_warning(
+                    f'Tried to add an AdEx event that already exists in the DB. '
+                    f'Event data: {event_tuple}. Skipping event.',
+                )
+                continue
+
+        self.conn.commit()
+        self.update_last_write()
+
+    def get_adex_events(
+            self,
+            from_timestamp: Optional[Timestamp] = None,
+            to_timestamp: Optional[Timestamp] = None,
+            address: Optional[ChecksumEthAddress] = None,
+            bond_id: Optional[HexStr] = None,
+            event_type: Optional[AdexEventType] = None,
+    ) -> List[Union[Bond, Unbond, UnbondRequest, ChannelWithdraw]]:
+        """Returns a list of AdEx events optionally filtered by time and address.
+        """
+        cursor = self.conn.cursor()
+        query = (
+            'SELECT '
+            'tx_hash, '
+            'address, '
+            'identity_address, '
+            'timestamp, '
+            'type, '
+            'pool_id, '
+            'amount, '
+            'usd_value, '
+            'bond_id, '
+            'nonce, '
+            'slashed_at, '
+            'unlock_at, '
+            'channel_id '
+            'FROM adex_events '
+        )
+        # Timestamp filters are omitted, done via `form_query_to_filter_timestamps`
+        filters = []
+        if address is not None:
+            filters.append(f'address="{address}" ')
+        if bond_id is not None:
+            filters.append(f'bond_id="{bond_id}"')
+        if event_type is not None:
+            filters.append(f'type="{event_type}"')
```

Comment:
This is a bug here. You are comparing directly with the event type. You need to turn it into a string.

Reply:
Fixed it. Please resolve when you read it.